src/endpoints/auth/auth.controller.ts

```diff
@@ -0,0 +1,95 @@
+import { FastifyPluginAsync, FastifyPluginOptions } from 'fastify';
+import { UserEntity, VerificationCodes } from '../../shared/database';
+import Ajv from 'ajv';
+const ajv = new Ajv();
+import jwt from 'jsonwebtoken';
+import { createHmac, randomUUID } from 'crypto';
+import { createTransport } from 'nodemailer';
+
+const secret = process.env.CRYPT_SECRET!;
+
+const generateTokens = (sessionKey: string) => {
+  const accessToken = jwt.sign({ sessionKey }, process.env.JWT_ACCESS_SECRET!, { expiresIn: '1d' });
+  const refreshToken = jwt.sign({ sessionKey }, process.env.JWT_REFRESH_SECRET!, { expiresIn: '31d' });
+
+  return { accessToken, refreshToken };
+};
+
+const createHash = (password: string, secretKey = secret): string => {
+  return createHmac('sha256', secretKey).update(password).digest('hex');
+};
+
+const transporter = createTransport({
+  host: String(process.env.SMTP_HOST),
+  port: Number(process.env.SMTP_PORT),
+  secure: false,
+  auth: {
+    user: process.env.SMTP_USER!,
+    pass: process.env.SMTP_PASSWORD!,
+  },
+});
+
+const signUpOptions = {
+  schema: {
+    body: {
+      200: {
+        type: 'object',
```

Comment:
Create `inputs` and `outputs` folders in this folder. Move models and interfaces there. This is an output.